eZ/Publish/Core/Persistence/Legacy/Content/Type/Handler.php

```diff
@@ -489,7 +489,7 @@ public function getFieldDefinition($id, $status)
     {
         $rows = $this->contentTypeGateway->loadFieldDefinition($id, $status);
 
-        if ($rows === false) {
+        if ($rows === false || (is_array($rows) && count($rows) === 0)) {
```

Comment:
For me as a fix this is ok, but maybe instead of handling two data types returned from `loadFieldDefinition` we should fix it that `     * @return array Data rows.` is always true. @alongosz @adamwojs ?

Reply:
It's already done. See https://github.com/ezsystems/ezpublish-kernel/pull/2975/files#diff-c3f4f47932e557d64e1452b72d7dc8a6R80

Reply:
Sorry, so should we assume `$rows` will always be an array? And just check if it has some items?

Reply:
@adamwojs

Reply:
This looks like some legacy debt. `loadFieldDefinition` will never return boolean, at least by PDO in PHP 7.x.
```suggestion
        if (count($rows) === 0) {
```